core/tests/test_utils.py

```diff
@@ -564,6 +596,38 @@ def tearDown(self):
     def _get_all_queue_names(self):
         return [q['name'] for q in self.taskqueue_stub.GetQueues()]
 
+    @contextlib.contextmanager
+    def urlfetch_mock(
+            self, content='', status_code=200, headers=None):
+        """Enables the custom urlfetch mock (URLFetchServiceMock) within the
+        context of a 'with' statement.
+
+        This mock is currently used for signup to prevent external HTTP
+        requests to fetch the Gravatar profile picture for new users while the
+        backend tests are being run.
+
+        args:
+          - content: Response content or body.
+          - status_code: Response status code.
+          - headers: Response headers.
+        """
+        if not headers:
```

Comment:
if headers is None (I think that's the specific case you're trying to catch, rather than e.g. {})